src/confd/src/ietf-interfaces.c

```diff
@@ -891,46 +885,47 @@ static int netdag_gen_veth(struct dagger *net, struct lyd_node *dif,
 static int netdag_gen_vlan(struct dagger *net, struct lyd_node *dif,
 			   struct lyd_node *cif, FILE *ip)
 {
-	const char *parent = lydx_get_cattr(cif, "parent-interface");
+	const char *lower_if;
 	const char *ifname = lydx_get_cattr(cif, "name");
 	struct lydx_diff typed, vidd;
-	struct lyd_node *otag;
+	struct lyd_node *vlan;
 	const char *proto;
 	int err;
 
-	DEBUG("ifname %s parent %s", ifname, parent);
-
-	err = dagger_add_dep(net, ifname, parent);
-	if (err)
-		return ERR_IFACE(cif, err, "Unable to add dep \"%s\"", parent);
-
-	otag = lydx_get_descendant(lyd_child(dif ? : cif),
-				   "encapsulation",
-				   "dot1q-vlan",
-				   "outer-tag",
+	vlan = lydx_get_descendant(lyd_child(dif ? : cif),
+				   "vlan",
 				   NULL);
-	if (!otag) {
+	if (!vlan) {
 		/*
-		 * Note: this is only an error if outer-tag is missing
+		 * Note: this is only an error if vlan subcontext is missing
 		 * from cif, otherwise it just means the interface had a
 		 * a change that was not related to the VLAN config.
 		 */
 		if (!dif)
-			ERROR("%s: missing mandatory outer-tag", ifname);
+			ERROR("%s: missing mandatory vlan", ifname);
 		return 0;
 	}
+	
+	lower_if = lydx_get_cattr(vlan, "lower-layer-if");
+
+	DEBUG("ifname %s lower if %s\n", ifname, lower_if);
+
+	err = dagger_add_dep(net, ifname, lower_if);
+	if (err)
+		return ERR_IFACE(cif, err, "Unable to add dep \"%s\"", lower_if);
```

Comment:
Maybe check `lower_if` before sending to dagger_add_dep() and `ERR_IFACE()`?

Reply:
As discussed AFK, no need to check `lower_if` here, the model ensures it has to be set.